nexus/db-queries/src/db/datastore/instance.rs

```diff
@@ -1321,22 +1321,7 @@ impl DataStore {
         .await
     }
 
-    /// Delete the provided `authz_instance`, as long as it is in one of the
-    /// provided set of [`InstanceState`]s.
-    ///
-    /// A.K.A. "[`project_delete_instance`], hard mode". Typically, instances
-    /// may only be deleted if they are in the [`InstanceState::NoVmm`] or
-    /// `[`InstanceState::Failed`] states. This method exists separately in
-    /// order to allow an unwinding `instance-create` saga to delete the
-    /// instance record without transiently moving it to
-    /// [`InstanceState::Failed`], in which it becomes eligible for automatic
-    /// restart.
-    ///
-    /// In general, callers should use [`project_delete_instance`] instead,
-    /// unless you really know what you're doing.
-    ///
-    /// [`project_delete_instance`]: Self::project_delete_instance
-    pub async fn project_delete_instance_in_states(
+    async fn project_idempotently_mark_instance_deleted_in_states(
```

Comment:
also, this is kind of a turbo nit, but, AFAICT, this function exists only to be called by `project_delete_instance_in_states`. personally, i like to lay out code in a sort of topological sorted order, with leaf functions like this defined after the public API functions that call them, so that as you read through the file, you see the one place where the function is called first, and then encounter its definition. so i would put this after `project_delete_instance_in_states`...although that might result in a larger diff...

Reply:
Happy to do that! I was leaving it the old way for the diff, but you've already reviewed this, so I'm happy to shuffle